src/urllib3/__init__.py

```diff
@@ -2,6 +2,36 @@
 Python HTTP library with thread-safe connection pooling, file post support, user friendly, and more
 """
 
```

Comment:
```suggestion
from __future__ import generator_stop  # urllib 3 needs python 3.6+
```

Reply:
@graingert We can't put this here because the top of the file needs to be valid Python 2.7, which this line isn't! This is an interesting option, but out of scope for this pull request

Reply:
The user will see the comment in the traceback

Reply:
eg like this:

```pytb
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "urllib3/__init__.py", line 1
    from __future__ import generator_stop  # urllib3 needs python 3.6+
SyntaxError: future feature generator_stop is not defined
```

Reply:
then when 3.6 is dropped you can upgrade to `from __future__ import annotations`

Reply:
I'd prefer to be explicit, if a library did this it wouldn't be immediately obvious to me that my Python version isn't supported.